src/app/icd/client/ICDClientInfo.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPConfig.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/ScopedNodeId.h>
+#include <lib/support/CodeUtils.h>
+#include <stddef.h>
+
+namespace chip {
+namespace app {
+
+class ICDClientInfo
+{
+public:
+    ScopedNodeId mPeerNode;
+    uint32_t mStartICDCounter          = 0;
+    uint32_t mOffset                   = 0;
+    uint64_t mMonitoredSubject         = static_cast<uint64_t>(0);
+    Crypto::Aes128KeyHandle mSharedKey = Crypto::Aes128KeyHandle();
```

Comment:
This has public members, like a struct. This should be a struct. Otherwise make the accessors for the class and keep members private.

If used as a POD struct, then rename members like "peer_node/start_icd_counter/monitored_subject", etc and remove the `m` (which is for private/protected members).

Please document what each field is for.

Reply:
updated with POD struct, thanks